### PR TITLE
fix AntTextArea autosize not working correctly

### LINKED ISSUE #34: [Input] Autosize not working correctly for TextArea
a0c50ae3-23e0-4202-adfa-6892f6cd3fa1

Maybe [this](https://www.jacklmoore.com/autosize/) script can help?

### PR DESCRIPTION
**Pull request by Brian-Ding**
fix #34

### CODE REVIEW

File: components/input/AntTextArea.razor.cs [hunk 1]
```diff
@@ -1,32 +1,100 @@
...
-        private RenderFragment _hidden;
+        /// <summary>
+        /// scrollHeight of 1 row
+        /// </summary>
+        private double _rowHeight;
+
+        /// <summary>
+        /// total height = row * _rowHeight + _offsetHeight
+        /// </summary>
+
```
Comment (lindexi) [inline components/input/AntTextArea.razor.cs]: 多余空行

File: components/input/AntTextArea.razor.cs [hunk 2]
```diff
@@ -1,32 +1,100 @@
...
     public partial class AntTextArea : AntInput
     {
-        private RenderFragment _hidden;
+        /// <summary>
+        /// scrollHeight of 1 row
+        /// </summary>
+        private double _rowHeight;
+
+        /// <summary>
+        /// total height = row * _rowHeight + _offsetHeight
```
Comment (lindexi) [inline components/input/AntTextArea.razor.cs]: 建议使用 `<see ` 的方式，这样后续更改变量名可以自动替换

File: components/input/AntTextArea.razor.cs [hunk 3]
```diff
@@ -1,32 +1,100 @@
...
-        public EventCallback<object> onResize { get; set; }
+        public uint MinRows
+        {
+            get
+            {
+                return _minRows;
+            }
+            set
+            {
+                if (value >= 1 && value <= MaxRows)
```
Comment (lindexi) [inline components/input/AntTextArea.razor.cs]: 是否将 1 设置为默认的最小值，添加一个常量

File: components/input/AntTextArea.razor.cs [hunk 4]
```diff
@@ -1,32 +1,100 @@
...
+                {
+                    _minRows = value;
+                    AutoSize = true;
+                }
+                else
+                {
+                    throw new ArgumentOutOfRangeException(nameof(MinRows));
```
Comment (lindexi) [inline components/input/AntTextArea.razor.cs]: 建议加上 message 内容，告诉开发者值的范围是多少，当前的值是多少

File: components/input/AntTextArea.razor.cs [hunk 5]
```diff
@@ -1,32 +1,100 @@
...
+        [Parameter]
+        public uint MaxRows
+        {
+            get
+            {
+                return _maxRows;
+            }
+            set
+            {
+                if (value <= uint.MaxValue && value >= MinRows)
```
Comment (lindexi) [inline components/input/AntTextArea.razor.cs]: 第一个判断应该是多余的（我真不知道能大于）

**Comment by vertonghenb** [comment]
To be honest I wouldn't use JS for this, it can be accomplished by CSS and pure C#.

**Comment by ElderJames** [comment]
Hi @vertonghenb ,
You can directly submit a PR.

**Comment by Brian-Ding** [comment]
@vertonghenb, i looked at your sample, but that required a extra container div out of textarea, I would like to keep the AntTextArea acting as a textarea element when rendered.

**Comment by ElderJames** [comment]
@Brian-Ding  Less js is better.

**Pull request merged**